enhancements/network/external-lb-vips.md

```diff
@@ -0,0 +1,213 @@
+   loadbalancer is updated and the DNS record has been updated and the results are propagated as well.
+
+### API Extensions
+
+A new field will be added to the `PlatformStatus` of the infrastructure object to tell
+the kind of the loadbalancer.
+
+```golang
+// LoadBalancerType can be either external or internal 
+type LoadBalancerType string
+const (
+	// LoadBalancerTypeExternal
+	LoadBalancerTypeExternal LoadBalancerType = "external"
+	// LoadBalancerTypeInternal
+	LoadBalancerTypeInternal LoadBalancerType = "internal"
+)
+// VSpherePlatformStatus holds the current status of the vSphere infrastructure provider.
+type VSpherePlatformStatus struct {
+  // loadBalancerType defines if a internal or external load balancer
+  // is to be used for the openshift api and ingress endpoints.
+  // When set to Internal the static pods defined in machine config operator
+  // will be used.
+  // When set to External the static pods will not be deployed. It will be 
+  // expected that the LB and DNS A records be pre-configured prior to installation.
+  // +kubebuilder:default=internal
+  // +kubebuilder:validation:Enum=internal;external
+  LoadBalancer LoadBalancerType `json:"loadBalancerType,omitempty"`
+}
+```
+
+This field will be set in the `install-config.yaml`:
+```yaml
+platform:
+  vsphere:
+    loadBalancerType: external
+```
+
+### Implementation Details/Notes/Constraints
+
+* CoreDNS will not be deployed if external loadbalancer is being used.
+  The customer is responsible of populating the DNS records for api, api-int and ingress.
+  We need to check who is responsible for machine records.
+
+* We need to make it ready for any platform who wants to support it, and allow
+  platforms to turn this on in the installer project. This will be achieved while validating the
+  installconfig manifest.
+
+* `PlatformStatus` won't have the VIP fields if LB type is external.
+
+
+### Risks and Mitigations
+
+Minimal risk, some platforms (vsphere, openstack) have been testing it already and no issue
+has been seen so far with this scenario.
+Validations will have to be implemented to make sure that the customer has prepare their
+environment before the IPI deployment.
+
+### Drawbacks
+
+This has been asked by customers, so they can have more flexibility so there is nothing to highlight here.
+
+## Design Details
+
+### Open Questions
+
+* Do we agree on not deploying CoreDNS if an external loadbalancer is being used?
+* Do we want to set the new field in `PlatformSpec` or `PlatformStatus` ?
+* Do we want to support a migration from internal loadbalancer to external loadbalancer?
+
+### Test Plan
+
+CI will be in charge of testing this feature.
+Platforms who want to support this will have their jobs.
+Note: vsphere already has a CI job for external loadbalancer. We'll work together to converge our tools.
+
+### Graduation Criteria
+
+The Tech-Preview target is OCP 4.13 and GA would be 4.14.
```

Comment:
We are 1.5 sprints from 4.13 being cut, do we realistically have time to get this feature in given we are still discussing the enhancement?

Reply:
I know this is stretch which is why I've been syncing with you guys so we can agree on a design for 4.13. The code has made good progress.
I don't know if it'll make it but I'm working full time on it anyway... We have large customers waiting for it.